gf/include/alps/gf/gf_base.hpp

```diff
@@ -448,6 +448,26 @@ namespace alps {
                                             [](VTYPE a, VTYPE b) {return std::abs(a) < std::abs(b);} ) );
         }
 
+        // reshape green's function
+        template<typename St = Storage>
```

Comment:
I don't like those artificial template parameters needed only to make `std::enable_if` work.
In this simple case you could go with something cleaner:
```c++
        // reshape green's function
        void reshape(MESHES...meshes) {
          std::array<size_t, N_> new_shape = get_sizes(std::make_tuple(meshes...));

          if(std::is_same<data_view, Storage>::value) {
            size_t new_size = std::accumulate(new_shape.begin(), new_shape.end(), size_t(1), std::multiplies<size_t>());
            if(data_.size() != new_size) {
              throw std::invalid_argument("The total size of resulting Green's function view should be the same as before.");
            }
          }

          data_.reshape(new_shape);
          meshes_ = std::make_tuple(meshes...);
          empty_ = false;
        }
```